src/controller.hpp

```diff
@@ -0,0 +1,81 @@
+/**
+ * @file
+ * @brief     This file contains a class for the motor controller
+ * @author    Nick Bout
+ * @license   MIT License
+ */
+
+#ifndef CONTROLLER_HPP
+#define CONTROLLER_HPP
+
+#include "dc_interface.hpp"
+#include "motor_interface.hpp"
+#include "stepper_interface.hpp"
+#include "wrap-hwlib.hpp"
+
+namespace MotorController {
+/**
+ * @brief Controller for motor logic
+ *
+ * This class is used to control a motor using the motor_interface superclass.
+ * The interface can be set to DC or STEPPER and uses the selected interface to set speed, direction or angle of a motor
+ *
+ */
+class Controller {
+  public:
+    ///< Enum class with all available motor interfaces
+    enum class INTERFACE { DC, STEPPER };

```

Comment:
Personally, an enum class should be named PascalCases, just like classnames. Fully caps names for macros are introduced due to macros being dangerous in C and required to be used with care. We are programming in C++ where the usage of macro's almost never is required, so there is no particular reason for them to be fully upper case in R2D2.